BitwardenShared/Core/Auth/Services/AuthService.swift

```diff
@@ -573,17 +578,55 @@ class DefaultAuthService: AuthService { // swiftlint:disable:this type_body_leng
             kdfParams: response.sdkKdf,
             purpose: .serverAuthorization
         )
-        let token = try await getIdentityTokenResponse(
-            authenticationMethod: .password(username: username, password: hashedPassword),
-            email: username,
-            captchaToken: captchaToken
-        )
+
+        var token: IdentityTokenResponseModel?
+        do {
+            token = try await getIdentityTokenResponse(
+                authenticationMethod: .password(username: username, password: hashedPassword),
+                email: username,
+                captchaToken: captchaToken
+            )
+        } catch let error as IdentityTokenRequestError {
+            if case let .twoFactorRequired(_, _, masterPasswordPolicyResponseModel, _) = error {
+                try await checkMasterPasswordPolicies(masterPassword, true, username, masterPasswordPolicyResponseModel)
+            }
+            // Re-throw the error.
+            throw error
+        }
```

Comment:
🎨 🤔 We're currently handling `IdentityTokenRequestError` inside `getIdentityTokenResponse` and even more we're also handling `.twoFactorRequired(...)` in there as well so in order to maintain that handling as centralized as possible I think this should be moved there. If the issue was that this is only required for login with master password, then just do an if check that the `authenticationMethod` is `.password` and then you can handle your specific case all in the same place.